src/services/DatasetService.js

```diff
@@ -1,58 +1,134 @@
 import datasette from './datasette.js'
 import logger from '../utils/logger.js'
+import { types } from '../utils/logging.js'
+import JSON5 from 'json5' // need to use this as spec is written with single quotes. need to pass this onto data designers team to fix
 import performanceDbApi from './performanceDbApi.js'
 
-async function getDatasetStatsForResourceId (dataset, resourceId) {
-  const sql = `
-    SELECT 'numberOfRecords' AS metric, COUNT(*) AS value
-    FROM
-      (
-        SELECT
-          *
-        FROM
-          fact_resource fr
-        WHERE
-          fr.resource = '${resourceId}'
-        GROUP BY
-          entry_number
-      )
-    UNION ALL
-    SELECT 'numberOfFieldsSupplied' AS metric, COUNT(*) AS value
-    FROM
-    (
-      SELECT
-        *
-      FROM
-        fact_resource fr
-      WHERE
-        fr.resource = '${resourceId}'
-    )`
-
-  const { formattedData } = await datasette.runQuery(sql, dataset)
+async function getColumnSummary (dataset, lpa) {
```

Comment:
I'd like to get rid of all functions like this and replace them with 

```js
fetchMany({
    query: (req) => TODO,
    result: 'columnSummary',
    dataset: FetchOptions.performanceDb
})
```

See how the `dataset` option is used (in retrospect the name clashes with the particular notion of 'dataset' of LPA, but that's for another time).

Note, the `FetchOptions.performanceDb` doesn't exist now, but you can add it and add  a condition in `datasetOverride` (it's one of the most common use cases, so it probably should have it's own option).